docs/dev_guide.md

```diff
@@ -1,31 +1,166 @@
+      the [testing guide](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/docs/dev_guide.md#testing-of-new-features)
+      below.
+
+* After the feature development is complete, developers should open a GitHub
+  issue (
+  see [example](https://github.com/GoogleCloudPlatform/gcsfuse/issues/793)) to
+  track the feature's progress towards production readiness. This issue should
+  outline the motivation for the feature, its design, and any known limitations.
+
+**2. Production Rollout:**
+
+* Once the feature has been thoroughly vetted in test environments and approved
+  by the GCSFuse team, it will be rolled out for general availability. The
+  GCSFuse team will own this part.
+* This involves:
+    * Removing the experimental prefix from the flag.
+    * Updating the Google Cloud documentation to include the new feature.
+
+## Adding a new CLI Flag/Config to GCSFuse
 
 Each param in GCSFuse should be supported as both as a CLI flag as well as via
 config file; the only exception being *--config-file* which is supported only as
 a CLI flag for obvious reasons. Please follow the steps below in order to make a
 new param available in both the modes:
 
-1.  Declare the new param in
-    [params.yaml](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/cfg/params.yaml#L4).
-    Refer to the documentation at the top of the file for guidance.
-1.  Run `make build` from the project root to generate the required code.
-1.  Add validations on the param value in
-    [validate.go](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/cfg/validate.go)
-1.  If there is any requirement to tweak the value of this param based on other
-    param values or other param values based on this one, such a logic should be
-    added in
-    [rationalize.go](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/cfg/rationalize.go).
-    If we want that when an int param is set to -1 it should be replaced with
-    `math.MaxInt64`, rationalize.go is the appropriate candidate for such a
-    logic.
-1.  Add unit tests in
-    [validate_test.go](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/cfg/validate_test.go)
-    and
-    [rationalize_test.go](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/cfg/rationalize_test.go)
-    and composite tests in
-    [config_validation_test.go](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/cmd/config_validation_test.go),
-    [config_rationalization_test.go](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/cmd/config_rationalization_test.go)
-    and
-    [root_test.go](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/cmd/root_test.go)
+1. Declare the new param in
+   [params.yaml](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/cfg/params.yaml#L4).
+   Refer to the documentation at the top of the file for guidance.
+2. Run `make build` from the project root to generate the required code.
```

Comment:
Why are we explicitly assigning indices instead of letting the renderer compute them itself. If we let it auto-compute by assigning index 1. for every item, we don't have to renumber while adding/removing steps.

Reply:
It was showing up as a warning on my IDE so I applied the suggested fix. I don't think this is an added toil as it is auto fixed by IDE. Let me know if you think this should be reverted.